app/controllers/api/v1/search_results_controller.rb

```diff
@@ -21,6 +18,16 @@ def create
 
       private
 
+      def create_upload_form
+        current_user = User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
```

Comment:
The usual way to do this is to define a method `current_user` in ApplicationController so that it can be re-used in all controllers. Example: https://github.com/nimblehq/nimble-survey-web/blob/3f8c20fdf704f7fb502f236a5087bd8500ca9fcd/app/controllers/concerns/api/v1/doorkeeper_authentication.rb#L16